tf2_sensor_msgs/test/test_tf2_sensor_msgs.py

```diff
@@ -29,61 +29,186 @@
 # POSSIBILITY OF SUCH DAMAGE.
 
 import copy
-import struct
 import unittest
 
-from sensor_msgs import point_cloud2
+from geometry_msgs.msg import Transform, TransformStamped
+import numpy as np
 from sensor_msgs.msg import PointField
-from tf2_ros import TransformStamped
-import tf2_sensor_msgs
+from sensor_msgs_py.point_cloud2 import create_cloud, create_cloud_xyz32, read_points_numpy
+from std_msgs.msg import Header
+import tf2_ros
+from tf2_sensor_msgs import do_transform_cloud, transform_points
 
 
-# A sample python unit test
 class PointCloudConversions(unittest.TestCase):
 
     def setUp(self):
-        self.point_cloud_in = point_cloud2.PointCloud2()
-        self.point_cloud_in.fields = [
-            PointField('x', 0, PointField.FLOAT32, 1),
-            PointField('y', 4, PointField.FLOAT32, 1),
-            PointField('z', 8, PointField.FLOAT32, 1)]
-
-        self.point_cloud_in.point_step = 4 * 3
-        self.point_cloud_in.height = 1
-        # we add two points (with x, y, z to the cloud)
-        self.point_cloud_in.width = 2
-        self.point_cloud_in.row_step = \
-            self.point_cloud_in.point_step * self.point_cloud_in.width
-
-        points = [1, 2, 0, 10, 20, 30]
-        self.point_cloud_in.data = struct.pack('%sf' % len(points), *points)
-
-        self.transform_translate_xyz_300 = TransformStamped()
-        self.transform_translate_xyz_300.transform.translation.x = 300
-        self.transform_translate_xyz_300.transform.translation.y = 300
-        self.transform_translate_xyz_300.transform.translation.z = 300
-        # no rotation so we only set w
-        self.transform_translate_xyz_300.transform.rotation.w = 1
+        self.points = np.array([[1, 2, 0], [10, 20, 30]], dtype=np.float32)
 
-        assert(list(point_cloud2.read_points(self.point_cloud_in)) ==
-               [(1.0, 2.0, 0.0), (10.0, 20.0, 30.0)])
+        self.point_cloud_in = create_cloud_xyz32(
+            Header(frame_id="test"),
+            self.points)
 
     def test_simple_transform(self):
-        # deepcopy is not required here because we have a str for now
+        k = 300.0
+        transform_translate_xyz = TransformStamped()
+        transform_translate_xyz.transform.translation.x = k
+        transform_translate_xyz.transform.translation.y = k
+        transform_translate_xyz.transform.translation.z = k
+        # no rotation so we only set w
+        transform_translate_xyz.transform.rotation.w = 1.0
+
+        # Check if point cloud contains our points
+        self.assertTrue(np.allclose(
+            read_points_numpy(self.point_cloud_in),
+            [(1.0, 2.0, 0.0), (10.0, 20.0, 30.0)]))
```

Comment:
Let's not put in this assertion.

By construction, functional code and test code are tested one against the other, and if the point cloud does not contain "our points", the tests won't pass anyway.